src/main/scala/com/databricks/spark/sql/perf/tpcds/Tables.scala

```diff
@@ -85,14 +79,70 @@ class Tables(sqlContext: SQLContext, dsdgenDir: String, scaleFactor: Int) extend
 
       val convertedData = {
         val columns = schema.fields.map { f =>
-          val columnName = new ColumnName(f.name)
-          columnName.cast(f.dataType).as(f.name)
+          col(f.name).cast(f.dataType).as(f.name)
         }
         stringData.select(columns: _*)
       }
 
       convertedData
     }
+
+    def genData(location: String, format: String, overwrite: Boolean): Unit = {
+      val mode = if (overwrite) SaveMode.Overwrite else SaveMode.Ignore
+
+      val withFormatAndMode = df.write.format(format).mode(mode)
+      val withPartitionColumns = if (partitionColumns.isEmpty) {
+        withFormatAndMode
+      } else {
+        withFormatAndMode.partitionBy(partitionColumns : _*)
+      }
+      logInfo(s"Generating table $name in database to $location with save mode $mode.")
+      withPartitionColumns.save(location)
+    }
+
+    def createExternalTable(location: String, format: String, databaseName: String, overwrite: Boolean): Unit = {
+      val qualifiedTableName = databaseName + "." + name
+      val tableExists = sqlContext.tableNames(databaseName).contains(name)
+      if (overwrite) {
+        sqlContext.sql(s"DROP TABLE IF EXISTS $databaseName.$name")
+      }
+      if (!tableExists || overwrite) {
+        logInfo(s"Creating external table $name in database $databaseName using data stored in $location.")
+        sqlContext.createExternalTable(qualifiedTableName, location, format)
+      }
+    }
+
+    def createTemporaryTable(location: String, format: String): Unit = {
+      logInfo(s"Creating temporary table $name using data stored in $location.")
+      sqlContext.read.format(format).load(location).registerTempTable(name)
+    }
+  }
+
+  def genData(location: String, format: String, overwrite: Boolean): Unit = {
+    tables.foreach { table =>
+      val tableLocation =
+        location + File.separator + format + File.separator + "sf" + scaleFactor + File.separator + table.name
+      table.genData(tableLocation, format, overwrite)
+    }
+  }
+
+  def createExternalTables(location: String, format: String, databaseName: String, overwrite: Boolean): Unit = {
+    sqlContext.sql(s"CREATE DATABASE IF NOT EXISTS $databaseName")
+    tables.foreach { table =>
+      val tableLocation =
+        location + File.separator + format + File.separator + "sf" + scaleFactor + File.separator + table.name
```

Comment:
maybe leave the specific location specification up to the user and just do `$location/${table.name}`